### PR TITLE
Fix nsmd k8s panic and re-registration error

### PR DESCRIPTION
**Pull request by alex-yust**
## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [x] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: controlplane/pkg/tests/nsmd_plugin_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/plugins"
+)
+
+func TestPluginReRegistration(t *testing.T) {
+	g := NewWithT(t)
+	var err error
+
+	const pluginName = "dummy-plugin"
+
+	registryDir := path.Join(os.TempDir(), "networkservicemesh")
```
[the remark this answers is not in the document]
Comment (alex-yust) [reply controlplane/pkg/tests/nsmd_plugin_test.go]: - [ ] cleanup the folder with sockets

File: controlplane/pkg/tests/nsmd_plugin_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		logrus.Errorf("Failed to create socket folder for Plugin Registry: %v", err)
+		return
+	}
+	defer func() { _ = os.RemoveAll(registryDir) }()
+
+	registryPath := path.Join(registryDir, path.Base(pluginsapi.PluginRegistrySocket))
+
+	pluginRegistry := plugins.NewPluginRegistry(registryPath)
+	if err = pluginRegistry.Start(context.Background()); err != nil {
+		logrus.Errorf("Failed to start Plugin Registry: %v", err)
```
Comment (lobkovilya) [inline controlplane/pkg/tests/nsmd_plugin_test.go]: `logrus.Errorf` doesn't fail the test. We should explicitly call `t.Fatalf("...reason...")` or something like that

File: controlplane/pkg/tests/nsmd_plugin_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+func TestPluginReRegistration(t *testing.T) {
+	g := NewWithT(t)
+	var err error
+
+	const pluginName = "dummy-plugin"
+
+	registryDir := path.Join(os.TempDir(), "networkservicemesh")
+	if err = os.MkdirAll(registryDir, os.ModeDir|os.ModePerm); err != nil {
+		logrus.Errorf("Failed to create socket folder for Plugin Registry: %v", err)
```
Comment (lobkovilya) [inline controlplane/pkg/tests/nsmd_plugin_test.go]: same here

File: controlplane/pkg/tests/nsmd_plugin_test.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	}()
+
+	plugin := &dummyConnectionPlugin{
+		prefixes: []string{"10.10.1.0/24"},
+	}
+	services := make(map[pluginsapi.PluginCapability]interface{}, 1)
+	services[pluginsapi.PluginCapability_CONNECTION] = plugin
+
+	if err = plugins.StartPlugin(pluginName, registryPath, services); err != nil {
+		logrus.Fatalln("Failed to start first instance of dummy plugin", err)
```
Comment (lobkovilya) [inline controlplane/pkg/tests/nsmd_plugin_test.go]: `logrus.Fatalln` will fail the test, but we won't get expected output like:
```
--- FAIL: TestPluginReRegistration (0.00s)
    nsmd_plugin_test.go:51: ...reason...
FAIL
```
so please use `t.Fatalf("...")` or similar methods

File: controlplane/pkg/tests/nsmd_plugin_test.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	time.Sleep(1 * time.Second)
+
+	if err = plugins.StartPlugin(pluginName, registryPath, services); err != nil {
+		logrus.Fatalln("Failed to start second instance of dummy plugin", err)
```
Comment (lobkovilya) [inline controlplane/pkg/tests/nsmd_plugin_test.go]: same here

File: controlplane/pkg/tests/nsmd_plugin_test.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+		prefixes: []string{"10.10.1.0/24"},
+	}
+	services := make(map[pluginsapi.PluginCapability]interface{}, 1)
+	services[pluginsapi.PluginCapability_CONNECTION] = plugin
+
+	if err = plugins.StartPlugin(pluginName, registryPath, services); err != nil {
+		logrus.Fatalln("Failed to start first instance of dummy plugin", err)
+	}
+
+	time.Sleep(1 * time.Second)
```
Comment (lobkovilya) [inline controlplane/pkg/tests/nsmd_plugin_test.go]: What will happen without sleep?

File: controlplane/pkg/tests/nsmd_plugin_test.go (hunk 2 above)
Comment (alex-yust) [reply controlplane/pkg/tests/nsmd_plugin_test.go]: - [ ] logrus.Errorf => logrus.Fatalf

File: controlplane/pkg/tests/nsmd_plugin_test.go (hunk 3 above)
Comment (alex-yust) [reply controlplane/pkg/tests/nsmd_plugin_test.go]: - [ ] logrus.Errorf => logrus.Fatalf

File: controlplane/pkg/tests/nsmd_plugin_test.go (hunk 5 above)
Comment (alex-yust) [reply controlplane/pkg/tests/nsmd_plugin_test.go]: - [ ] logrus.Fatalln => logrus.Fatalf

File: controlplane/pkg/tests/nsmd_plugin_test.go (hunk 4 above)
Comment (alex-yust) [reply controlplane/pkg/tests/nsmd_plugin_test.go]: - [ ] logrus.Fatalln => logrus.Fatalf

File: controlplane/pkg/tests/nsmd_plugin_test.go (hunk 6 above)
Comment (alex-yust) [reply controlplane/pkg/tests/nsmd_plugin_test.go]: - [ ] remove Sleep

File: controlplane/pkg/tests/nsmd_plugin_test.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+	if err = plugins.StartPlugin(pluginName, registryPath, services); err != nil {
+		t.Fatalf("Failed to start first instance of dummy plugin: %v", err)
+	}
+
+	if err = plugins.StartPlugin(pluginName, registryPath, services); err != nil {
+		t.Fatalf("Failed to start second instance of dummy plugin: %v", err)
+	}
+
+	g.Expect(err).To(BeNil())
```
Comment (denis-tingajkin) [inline controlplane/pkg/tests/nsmd_plugin_test.go]: [minor] `err` is nil always here

Comment (alex-yust) [reply controlplane/pkg/tests/nsmd_plugin_test.go]: Is it Ok to leave it here as a 'safety check'?

Comment (alex-yust) [reply controlplane/pkg/tests/nsmd_plugin_test.go]: - [ ] remove g.Expect

**Pull request merged**